yjit/src/codegen.rs

```diff
@@ -286,50 +299,42 @@ macro_rules! counted_exit {
 
 
 
-// FIXME: there is no longer a cb on the JITState object, so these
-// should take a &mut CodeBlock argument instead of JITState
-// Should also rename to be no longer jit_*
-//
-/*
 // Save the incremented PC on the CFP
-// This is necessary when calleees can raise or allocate
-static void
-jit_save_pc(jitstate_t *jit, x86opnd_t scratch_reg)
+// This is necessary when callees can raise or allocate
+fn jit_save_pc(jit: &JITState, cb: &mut CodeBlock, scratch_reg: X86Opnd)
 {
-    codeblock_t *cb = jit->cb;
-    mov(cb, scratch_reg, const_ptr_opnd(jit->pc + insn_len(jit->opcode)));
-    mov(cb, mem_opnd(64, REG_CFP, offsetof(rb_control_frame_t, pc)), scratch_reg);
+    let pc: *mut VALUE = jit.get_pc();
+    let ptr: *mut VALUE = unsafe { pc.offset(insn_len(jit.get_opcode()).try_into().unwrap()) };
```

Comment:
Here I would break this expression down into parts so it's more clear what's happening. For example `let cur_insn_len = ...`.

Reply:
Done.